sqlmesh/core/engine_adapter/trino.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import annotations
+
+import typing as t
+
+from sqlglot import exp
+from sqlglot.optimizer.qualify_columns import quote_identifiers
+
+from sqlmesh.core.dialect import to_schema
+from sqlmesh.core.engine_adapter.base import (
+    CatalogSupport,
+    InsertOverwriteStrategy,
+    SourceQuery,
+)
+from sqlmesh.core.engine_adapter.mixins import (
+    GetCurrentCatalogFromFunctionMixin,
+    HiveMetastoreTablePropertiesMixin,
+    LogicalReplaceQueryMixin,
+    PandasNativeFetchDFSupportMixin,
+)
+from sqlmesh.core.engine_adapter.shared import DataObject, DataObjectType
+
+if t.TYPE_CHECKING:
+    from trino.dbapi import Connection as TrinoConnection
+
+    from sqlmesh.core._typing import SchemaName, TableName
+
+
+class TrinoEngineAdapter(
+    PandasNativeFetchDFSupportMixin,
+    LogicalReplaceQueryMixin,
+    HiveMetastoreTablePropertiesMixin,
+    GetCurrentCatalogFromFunctionMixin,
+):
+    DIALECT = "trino"
+    ESCAPE_JSON = False
+    INSERT_OVERWRITE_STRATEGY = InsertOverwriteStrategy.INTO_IS_OVERWRITE
```

Comment:
what does this property mean

Reply:
In Trino we basically set a configuration that tells it to treat an `INTO` as if it was an `OVERWRITE`. Therefore we set this strategy and it tells the base adapter to just pretend like we are doing an overwrite while keeping `INTO`.

https://github.com/TobikoData/sqlmesh/blob/bd71fffd1051111ff5f2442fb29324b61124b88d/sqlmesh/core/engine_adapter/trino.py#L58-L64